### ISSUE TITLE
Gun Crafting/Drug Rework

### ISSUE THREAD

**Issue by ImArteria** [InZidiuZ/op-framework-issue-tracker#2272]
**Assurance:**
Have you read through the rules from the `README.md` file in the root folder of this GitHub repository?
Yes, Pog Frog.

**Summary:**
**THERE IS A TLDR FOR BEN (INZIDIUZ) AT THE END BECAUSE SIA YELLED AT ME LAST TIME**

Before getting into the main topic of it all, just want to note that essentially Guns/Drugs would be getting a complete overhaul/rework to allow groups who use these processes, have more time to RP. 

**Crafting as a Whole**

- A new property would be introduced for realtors to sell, called Warehouses. These warehouses apply purely to the criminal aspect of Legacy. These warehouses have a few different benefits but can be somewhat costly (intended purpose is for multiple people to own it)
          - Weapons/Drugs that are stored in a warehouse, have a 25% natural degrading reduction (meaning they last slightly longer)
          - There is a crafting bench in this warehouse that is used to craft drugs, guns, and attachments.
                    - When opening this crafting bench, it would look something like a PD locker (probably like a 5x5 Square), which you place different objects in to create new ones

**How to get Started (From a new Player's Perspective)**
The first thing every new player is told to do, is go "Do garbage" or find a job at lifeinvader. In these jobs at lifeinvader, there is a small possibility to get a "Empty Book" as a reward for completing these jobs. This empty book would be a usable items that introduces new players into crafting. They would have a few starter recipes (such as Advanced Lockpicks, Lockpicks, Repair kits, etc) and generally teaches them how to get into it. It would look like something like this (as linked, and I made it myself)

[Basic Needs](https://i.imgur.com/0wwzh53.png)

I did not photoshop this on this image, but in the bottom right would be an arrow that indicates to flip pages, followed by empty pages. As you may have guessed, these empty pages are reserved for Weapons/Drugs/Attachments! 

The next step, for criminals, would revolve around getting these blueprints. There would be many blueprints to prevent someone collecting them quickly and avoiding roleplay. To follow along with crafting, here's an example of a Weapons/Drugs pages that has some blueprints filled in, and some not.

[Drugs](https://i.imgur.com/W4HpK8K.png)
[Weapons](https://i.imgur.com/WuyfAio.png)

**Weapons**
To get the blueprint for weapons, this could be an affective way for it to happen(but not both, dev's choice W)

- At the old gun running location, you will now speak to a person who requests gun parts from you, in order for him to provide you a blueprint, you need to turn in gun parts which could be acquired by scrapping old guns, certain jobs, robbing houses, etc.
- Most likely, you would need a couple different spots for this quest to happen so people don't "camp" these quest spots, or it could possibly rotate
- Quest would be repeatable, and give one random blueprint (From like stock, grip, magazine, etc)
- You would than use this Blueprint, and it would add it to your book.

- Through the above, this will have some RP to occur between people trading parts in order to craft parts they need. An example scenario I listed on my document was like such
             - Person A: "Hey I have a magazine Blueprint and I need a grip, do you have it?"
             - Person B: "Yes I do, Ill craft you a grip if you craft me a magazine"
- Things like that would occur, and groups like Cartels could still make sense in RP because they could either provide blueprints for guns, parts, the guns themselves, etc. 
- Obviously you would take the parts, and have to fill in the recipe in the Warehouse.

**Cocaine/Process to Make it**
- Person would find Cocoa leaves during Trash runs, robbing houses, etc.
- They would go through the same process as weed, and grow it in their house.
- Once they acquire enough leaves, they need to soak it in gasoline. 
- They than need some chemicals which could be acquired in numerous ways, whichever Devs see fit (Whether it's a market, from jobs, or found naturally) and combine it to make Cocaine Paste. 
- This paste needs to dry, and they combine it with plastic and duct tape to make a brick (At the warehouse)

To get the blueprints for the above process, similar to guns, the person at the hanger would give them a quest to "Bring them 32 leaves", and they give them the blueprint for "Gasoline Soaked Cocoa Leaves". They than are given the quest to bring them these gasoline soaked leaves, given the next blueprint, etc. It goes until the process is completed and essentially teaches individuals how to make cocaine if they have the knowledge of it. 

**Oxy/Process to Make it**

- Person would acquire chemicals (same way to acquire Coke chemicals), which Im unsure of what exactly it is (I didn't research that much) and combine them. They would compress this into a pill, which would be further compressed into a bottle. Overall, the process would look like something below
- Acquire chemicals
- Mesh Chemicals into a pill
- Mesh multiple pills into a bottle

The blueprints for the above would follow the same type of way, would approach the Oxy door, would be somewhat "instructed" on how to get them (without revealing where you get the chemicals, etc) and teaches players how to make it. 

**Other Drugs**
- The above two processes follow the same pathline as Meth and future drugs. Essentially there would be a spot to speak to someone, they would basically teach players how to make these products, introduce crafting to them, and still provide roleplay. 

**TLDR FOR BEN**
- Warehouse, 25% less naturally degradation, crafting bench
- Doing legal jobs, you find a book. Has basic recipes for legal people and teaches people how to craft
- At certain locations, you can receive quests or "tips" on how to acquire more pages for this book
- This book has weapons, attachments, and drug pages missing
- These quests/tips teach players how to craft weapons, drugs, attachments
- Coke/Oxy/Meth different ways to get it.
- Overall, accomplishes the goal of having a healthy economy, without it being too hard or too easy.

**Reason:**
- Essentially, Groups who import the High Processes (Oxy, Guns, Coke) have to have somewhat of a "grind" mentality doing these processes, which to be fair comes with holding it down. This would be very beneficial towards RP because groups can still specialize in these processes, while it is available to all groups and focuses on Roleplay more as opposed to doing a mission for them.
- This system teaches new players, without handing them all the information, and allows new players to feel welcomed as they are taught the new system, but still need to roleplay out where these quests are, how to get items, how to get the book, and how to craft things. 
- This warehouse would also add roleplay by giving groups a "home" as one large building, while still maintaining a block and whatnot. This would need to be monitored by government managers to not allow one realtor to have multiple warehouses, and Group Managers so gangs don't have multiple. 
- Crafting is a sought-after thing, because it adds more roleplay and focuses on interaction more as opposed to doing missions to provide some RP down the line.

**Comment by coalaura** [comment]
The whole gun part i feel like has been done, but in a different way with the recent changes. Leaving this open for the drug changes

**Comment by ImArteria** [comment]
The above is a little bit outdated obviously, so with the gun portion being scratched I'd just figure I'll put a comment here of all the "new changes" purely just for drugs, reworked for the system we have now.

Personally I think we should move from a some processes being purely held by one group to a more universal approach due to some complications with them but for that purpose I'll be leaving Coke out out of the mix and suggest it stays as a "process" that can be held purely by one group.

The main few "drugs" that I want to cover is Oxy, Crack, and Heroine. Since weed has been recently changed I tended to leave that alone along with Meth.

**Black Market**
The black market would essentially be the "go to" place to go for crafting some of these drugs. Personally I recommend this location
https://i.imgur.com/r5bDrjZ.jpeg

Here you can buy various chemicals, "pharmacy bottles" and etc to create some of the above. 

**Crack**
Crack is typically made from cocaine, water, and some chemical like ammonia or baking soda so the process here would be relatively simple.
Cocaine is received from whichever dealer who is running the cocaine process for the server. Individuals have a choice to obviously sell coke raw, or go to the following locations to craft Crack.
Water - From the store obviously
Ammonia - From the black market dealer 
OR (Can be either Ammonia or Baking Soda)
Baking Soda - Purchased at some other location, perhaps where Yeast Seeds are purchased or a new location
Items can be crafted into crack at Cayo Perico in the hanger - https://i.imgur.com/0bmFKtH.png
Crack Cocaine can be sold in the same places as Cocaine


**Oxy**
Since Oxy needs an entire overhaul anyways, I wanted to recommend a different route for Oxy/Heroine.
Oxy is typically created from an Opium and obviously made into a pill-like form along with being placed into bottles. (Unfortunately google didn't really help me out here too much besides "It's made from an Opium from Poppy" so if others are able to find more specifically how it's crafted feel free to comment

At the location where Yeast Packets are purchased, "Poppy Seeds" are also able to be purchased. When purchasing these seeds, you are able to plant them on any "dirt" grounded area not blocked by a prop. Poppy would take roughly some amount of time to grow and are planted outside. The steps would detail as below
- Acquire seeds, plant outside, water every 3 hours or so, receive product

After you acquire the product, visit the black market, purchase bottles (like the actual oxy "bottle"), and water. You would than come to Pillbox where some small mapping work would need to be done.
This door would able to be opened - https://i.imgur.com/cM0ShPc.png
To which a pill press is found in here - https://i.imgur.com/wP7yMUL.png
Place the items in, and create your Oxy. 

I would also recommend that Oxy's "use" is changed overall. Personally how I describe the ideal Oxy affect is like Juggernaut from COD zombies. It doesn't heal you, it in fact just decreases incoming damage by 20-25% for the next 5 minutes. That way it wouldnt need to be mass created (even though it will anyway) and the value will increase astronomically. It also would make sense if you ODed off them more since you have no reason to pop more than 1 oxy at a time.


**Morphine**
Same process as Oxy for the most part. Planting poppy, etc. Unsure if we have a "microscope" prop but the Poppy would need to be extracted, so instead of creating Oxy for use of it, Morphine can be made by bringing Poppy to the same room as creating Oxy, to extract Morphine. The morphine would than need to be mixed with water, and at least the way google described this next step was "organic solvent" which google also said alcohol was included in that so sure why not!
- Alcohol, water, the Morphine extract is what is needed to create the total product.
- A syringe would need to be purchased from Mount Zonah where they sell the Medkits
- Morphine can be crafted into the syringe form at the Dam - https://i.imgur.com/rg938IQ.png

Product can be sold on North Rockford Drive and at the high school.




That's all I have for now. With these changes it would encourage groups to have "grow farms" to be able to create a commonly used product (assuming Oxy affect changes go through) or gain money through creating Morphine. Also Crack for the people who roleplay to get coke and create a better product.

**Comment by Milan345** [comment]
Can a config file be added?? with options to change (the location, required nos of crafting items etc)

why am I asking for these options particularly??
- it will prevent metagaming between subdivisions
- it will enhance rp between the crims & law enforcement. 
E.g. Law enforcement will be able to investigate on the source & on finding out the location, they will be able to seized that place. With a config file it would be easy to move location.

**Comment by coalaura** [comment]
> Can a config file be added?? with options to change (the location, required nos of crafting items etc)
> 
> 
> 
> why am I asking for these options particularly??
> 
> - it will prevent metagaming between subdivisions
> 
> - it will enhance rp between the crims & law enforcement. 
> 
> E.g. Law enforcement will be able to investigate on the source & on finding out the location, they will be able to seized that place. With a config file it would be easy to move location. 

No, thats a huge pain to implement and really not needed

**Comment by coalaura** [comment]
This part of the original suggestion was nice too

- A new property would be introduced for realtors to sell, called Warehouses. These warehouses apply purely to the criminal aspect of Legacy. These warehouses have a few different benefits but can be somewhat costly (intended purpose is for multiple people to own it)
- Weapons/Drugs that are stored in a warehouse, have a 25% natural degrading reduction (meaning they last slightly longer)
- There is a crafting bench in this warehouse that is used to craft drugs, guns, and attachments.
- When opening this crafting bench, it would look something like a PD locker (probably like a 5x5 Square), which you place different objects in to create new ones

**Comment by ImArteria** [comment]
> This part of the original suggestion was nice too
> 
>     * A new property would be introduced for realtors to sell, called Warehouses. These warehouses apply purely to the criminal aspect of Legacy. These warehouses have a few different benefits but can be somewhat costly (intended purpose is for multiple people to own it)
> 
>     * Weapons/Drugs that are stored in a warehouse, have a 25% natural degrading reduction (meaning they last slightly longer)

Ye I mean you know my opinion on them. I'd love for them to be the home base for storing illegal items, crafting big things, etc. Criminal Enterprise's homebase if you will, or a storage unit

**Comment by vol5** [comment]
@coalaura 

Warehouses how you just commented on here would alone be insane. But the idea of blueprints is very intriguing to me..

- The next step, for criminals, would revolve around getting these blueprints. There would be many blueprints to prevent someone collecting them quickly and avoiding roleplay. To follow along with crafting, here's an example of a Weapons/Drugs pages that has some blueprints filled in, and some not.

If you are cool with going down the crafting route T. Blueprints would be one hell of a way to add additional layers of content for players. The gangs would love that since it would provide a sense of adventure and progression they can work towards and its an idea that most gamers are familiar with. I like the idea of blueprints for the simple reason that its a phenomenal way to add seasonal/drip-feed content that benefits the groups in particular while being scalable. 

@ImArteria These are some damn good Ideas here....

**Comment by coalaura** [comment]
this issue is kinda outdated and very confusing, please create a new issue if parts of this issue are still relevant
